### PR TITLE
Fail fast when LitAPI.setup has error

### LINKED ISSUE #347: LitServe server doesn't crash when there is a failure in setup
### To Reproduce

Attach a [Lightning Studio](https://lightning.ai/studios) which is fully reproducible (code, dependencies, environment, etc...) to reproduce this:   

1. Create a [Studio](https://lightning.ai/studios).    
2. Reproduce the issue in the Studio.    
3. [Publish the Studio](https://lightning.ai/docs/overview/studios/publishing#how-to-publish).
4. Paste the Studio link here.    

```python
from litserve import LitAPI, LitServer

class DebugLitAPI(LitAPI):
    def setup(self, device):
        raise Exception

    def predict(self, inputs):
        return None

if __name__ == "__main__":
    api = DebugLitAPI()
    server = LitServer(api, accelerator='auto', devices='auto', workers_per_device=2)
    server.run(port=8000)
```

Instead, it should return exit code error 1.

### Environment
If you published a Studio with your bug report, we can automatically get this information. Otherwise, please describe:   

- PyTorch/Jax/Tensorflow Version (e.g., 1.0):
- OS (e.g., Linux):
- How you installed PyTorch (`conda`, `pip`, source):
- Build command you used (if compiling from source):
- Python version:
- CUDA/cuDNN version:
- GPU models and configuration:
- Any other relevant information:

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes #347 

LitServe doesn't crash when there is an error in the `LitAPI.setup` method. This PR makes LitServe fail when setup has error. 

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/litserve/loops.py [hunk 1]
```diff
@@ -399,18 +399,23 @@ def inference_worker(
     max_batch_size: int,
     batch_timeout: float,
     stream: bool,
-    workers_setup_status: Dict[str, bool],
+    workers_setup_status: Dict[int, str],
```
Comment (Borda) [inline src/litserve/loops.py:402]: can we have docs about the meaning or samples?

File: src/litserve/utils.py [hunk 2]
```diff
@@ -78,3 +79,11 @@ async def call_after_stream(streamer: AsyncIterator, callback, *args, **kwargs):
         logger.exception(f"Error in streamer: {e}")
     finally:
         callback(*args, **kwargs)
+
+
+@dataclasses.dataclass
+class WorkerSetupStatus:
```
Comment (Borda) [inline src/litserve/utils.py:85]: lets use strEnum https://github.com/Lightning-AI/utilities/blob/main/src/lightning_utilities/core/enums.py

File: src/litserve/loops.py (hunk 1 above)
Comment (aniketmaurya) [reply src/litserve/loops.py:402]: did you mean docstrings?

File: src/litserve/utils.py (hunk 2 above)
Comment (aniketmaurya) [reply src/litserve/utils.py:85]: what would be the benefit of using Enum over here?

**Review by ethanwharris — APPROVED** [review]

File: src/litserve/loops.py (hunk 1 above)
Comment (Borda) [reply src/litserve/loops.py:402]: Yes

File: src/litserve/utils.py (hunk 2 above)
Comment (Borda) [reply src/litserve/utils.py:85]: Becase you are using the data class as enum already

File: src/litserve/loops.py (hunk 1 above)
Comment (aniketmaurya) [reply src/litserve/loops.py:402]: great suggestion! although this is a private function so users won't be using it directly but might be useful for contributors.

**Pull request merged**